File: SECOM_AJIS.Presentation.Contract/Models/CTS220.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ComponentModel.DataAnnotations;

using SECOM_AJIS.Common.Util;
using SECOM_AJIS.Common.CustomAttribute;
using SECOM_AJIS.Presentation.Contract.Models.MetaData;
using SECOM_AJIS.Presentation.Contract.Models;
using SECOM_AJIS.Common.Models;
using SECOM_AJIS.DataEntity.Contract;
using SECOM_AJIS.DataEntity.Common;
using SECOM_AJIS.DataEntity.Quotation;
using SECOM_AJIS.DataEntity.Master;

namespace SECOM_AJIS.Presentation.Contract.Models
{
    /// <summary>
    /// Parameter of CTS220 screen
    /// </summary>
    public class CTS220_Parameter
    {
        public string contractCode { get; set; }
    }

    /// <summary>
    /// DO of Employee
    /// </summary>
    public class CTS220_DOEmployee : tbm_Employee
    {
        public string EmpNo { get; set; }
        public string EmpName { get; set; }
    }

    /// <summary>
    /// DO of ProductName
    /// </summary>
    public class CTS220_DOProductName : tbm_Product
    {
        [LanguageMappingAttribute]
        public string ProductName { get; set; }
    }

    /// <summary>
    /// DO of SelectProcess
    /// </summary>
    [MetadataType(typeof(CTS220_DOSelectProcess_Meta))]
    public class CTS220_DOSelectProcess
    {
        public string ContractCode { get; set; }
        public string OCC { get; set; }
        public string ProcessType { get; set; }
    }

    /// <summary>
    /// DO for validate business
    /// </summary>
    public class CTS220_ValidateBusiness
    {
        public string ContractCode { get; set; }
        public string OCC { get; set; }

        public string ContractFeeCurrencyType { get; set; }
        public decimal? ContractFee { get; set; }
        public string ContractFeeOnStopCurrencyType { get; set; }
        public decimal? ContractFeeOnStop { get; set; } //Add by Jutarat A. on 14082012
        public string ProcessType { get; set; }
        public DateTime? ChangeImplementDate { get; set; }
        public string ChangeType { get; set; }
        public string ChangeReasonType { get; set; }
        public string SecurityMemo { get; set; }
        //public string OperationTypeCode { get; set; }
        public string InsuranceTypeCode { get; set; }

        public string InsuranceCoverageAmountCurrencyType { get; set; }
        public decimal? InsuranceCoverageAmount { get; set; }
        public string MonthlyInsuranceFeeCurrencyType { get; set; }
        public decimal? MonthlyInsuranceFee { get; set; }

        public string DocumentCode { get; set; }
        public string PlanCode { get; set; }

        public string MaintenanceFee1CurrencyType { get; set; }
        public decimal? MaintenanceFee1 { get; set; }
        public string AdditionalFee1CurrencyType { get; set; }
        public decimal? AdditionalFee1 { get; set; }
        public string AdditionalFee2CurrencyType { get; set; }
        public decimal? AdditionalFee2 { get; set; }
        public string AdditionalFee3CurrencyType { get; set; }
        public decimal? AdditionalFee3 { get; set; }

        public string InstallationTypeCode { get; set; }
        public DateTime? InstallationCompleteDate { get; set; }

        public string NormalInstallFeeCurrencyType { get; set; }
        public decimal? NormalInstallFee { get; set; }
        public string OrderInstallFeeCurrencyType { get; set; }
        public decimal? OrderInstallFee { get; set; }
        public string OrderInstallFee_ApproveContractCurrencyType { get; set; }
        public decimal? OrderInstallFee_ApproveContract { get; set; }
        public string OrderInstallFee_CompleteInstallCurrencyType { get; set; }
        public decimal? OrderInstallFee_CompleteInstall { get; set; }
        public string OrderInstallFee_StartServiceCurrencyType { get; set; }
        public decimal? OrderInstallFee_StartService { get; set; }

        public DateTime? ReturnToOriginalFeeDate { get; set; }
        public DateTime? ExpectedResumeDate { get; set; }
        public string SalesmanEmpNo1 { get; set; }
        public string SalesmanEmpNo2 { get; set; }
        public string NegotiationStaffEmpNo1 { get; set; }
        public string NegotiationStaffEmpNo2 { get; set; }
        public string ApproveNo1 { get; set; }
        public string ApproveNo2 { get; set; }
        public bool IsSendNotifyEmail { get; set; }

        public bool IsEnableReasonType { get; set; }
        public List<string> OperationType { get; set; }

        public string NormalContractFeeCurrencyType { get; set; }
        public decimal? NormalContractFee { get; set; } //Add by Jutarat A. on 06022014

        public string InstallFeePaidBySECOMCurrencyType { get; set; }
        public decimal? InstallFeePaidBySECOM { get; set; }
        public string InstallFeeRevenueBySECOMCurrencyType { get; set; }
        public decimal? InstallFeeRevenueBySECOM { get; set; }
    }

    /// <summary>
    /// DO of Product
    /// </summary>
    public class CTS220_DOProductInformation
    {
        public string OrderContractFeeCurrencyType { get; set; }
        public string OrderContractFee { get; set; }
        public string ContractFeeOnStopCurrencyType { get; set; }
        public string ContractFeeOnStop { get; set; } //Add by Jutarat A. on 14082012
        public DateTime? ChangeImplementDate { get; set; }
        public string SecurityTypeCode { get; set; }
        public string ProductCode { get; set; }
        public string ChangeType { get; set; }
        public string ChangeReasonType { get; set; }
        public string SaleManEmpNo1 { get; set; }
        public string SaleManEmpName1 { get; set; }

        public string SaleManEmpNo2 { get; set; }
        public string SaleManEmpName2 { get; set; }

        public string ApproveNo1 { get; set; }
        public string ApproveNo2 { get; set; }
        public DateTime? CreateDate { get; set; }
        public string CreateBy { get; set; }
        public string CreateByName { get; set; }
        public string SecurityMemo { get; set; }
        public string OperationTypeCode { get; set; }
    }

    /// <summary>
    /// DO of InsuranceType
    /// </summary>
    public class CTS220_DOInsuranceType
    {
        public string InsuranceTypeCode { get; set; }
        public string InsuranceCoverageAmountCurrencyType { get; set; }
        public string InsuranceCoverageAmount { get; set; }
        public string MonthlyInsuranceFeeCurrencyType { get; set; }
        public string MonthlyInsuranceFee { get; set; }
    }

    /// <summary>
    /// DO of FutureDate
    /// </summary>
    public class CTS220_DOFutureDate
    {
        public DateTime? ExpectedResumeDate { get; set; }
        public DateTime? ReturnToOriginalFeeDate { get; set; }
    }

    /// <summary>
    /// DO of Quotation
    /// </summary>
    public class CTS220_DOQuotation
    {
        public string QuotationTargetCode { get; set; }
        public string QuotationAlphabet { get; set; }
        public string PlanCode { get; set; }
        public DateTime? PlanApproveDate { get; set; }
        public string PlanApproverEmpNo { get; set; }
        public string PlanApproverEmpName { get; set; }
        public string NormalContractFeeCurrencyType { get; set; }
        public string NormalContractFee { get; set; }
        public string MaintenanceFee1CurrencyType { get; set; }
        public string MaintenanceFee1 { get; set; }
        public string AdditionalFee1CurrencyType { get; set; }
        public string AdditionalFee1 { get; set; }
        public string AdditionalFee2CurrencyType { get; set; }
        public string AdditionalFee2 { get; set; }
        public string AdditionalFee3CurrencyType { get; set; }
        public string AdditionalFee3 { get; set; }
    }

    /// <summary>
    /// DO of Installation
    /// </summary>
    public class CTS220_DOInstallation
    {
        public string InstallationTypeCode { get; set; }
        public DateTime? InstallationCompleteDate { get; set; }

        public string NormalInstallFeeCurrencyType { get; set; }
        public string NormalInstallFee { get; set; }
        public string OrderInstallFeeCurrencyType { get; set; }
        public string OrderInstallFee { get; set; }
        public string OrderInstallFee_ApproveContractCurrencyType { get; set; }
        public string OrderInstallFee_ApproveContract { get; set; }
        public string OrderInstallFee_CompleteInstallCurrencyType { get; set; }
        public string OrderInstallFee_CompleteInstall { get; set; }
        public string OrderInstallFee_StartServiceCurrencyType { get; set; }
        public string OrderInstallFee_StartService { get; set; }
        public string InstallFeePaidBySECOMCurrencyType { get; set; }
        public string InstallFeePaidBySECOM { get; set; }
        public string InstallFeeRevenueBySECOMCurrencyType { get; set; }
        public string InstallFeeRevenueBySECOM { get; set; }
        public string InstallationSlipNoCurrencyType { get; set; }
        public string InstallationSlipNo { get; set; }
    }

    /// <summary>
    /// DO of SubContractor
    /// </summary>
    [MetadataType(typeof(CTS220_DOSubContract_Meta))]
    public class CTS220_DOSubContract
    {
        public string SubContractCode { get; set; }
        public string ProcessType { get; set; }
    }

    /// <summary>
    /// Parameter of CTS220 screen
    /// </summary>
    public class CTS220_ScreenParameter : ScreenSearchParameter
    {
        [KeepSession]
        public string ContractCode { get; set; }

        [KeepSession]
        public CTS220_Parameter ScreenParameter { get; set; }

        public CTS220_ValidateBusiness DOValidateBusiness { get; set; }
        public dsRentalContractData DSRentalContract { get; set; }
        public dsRentalContractData DSRentalContractPrevious { get; set; }
        public dsRentalContractData DSRentalContractNext { get; set; }
        public dsRentalContractData DSRentalContractShow { get; set; }

        public List<tbt_RentalSecurityBasic> ListRentalSecurityBasic { get; set; }
        public List<dtTbt_RentalInstSubContractorListForView> ListSubContractor { get; set; }
        public bool IsSendNotifyEmail { get; set; }

        public string ProcessType { get; set; }
    }

    /// <summary>
    /// DO for validate require field
    /// </summary>
    [MetadataType(typeof(CTS220_ValidateRequireField_MetaData))]
    public class CTS220_ValidateRequireField
    {
        public DateTime? ChangeImplementDate { get; set; }
        public string SalesmanEmpNo1 { get; set; }
    }

    /// <summary>
    /// DO for validate ChangeReasonType
    /// </summary>
    [MetadataType(typeof(CTS220_ValidateChangeReasonType_MetaData))]
    public class CTS220_ValidateChangeReasonType
    {
        public string ChangeReasonType { get; set; }
    }
}

namespace SECOM_AJIS.Presentation.Contract.Models.MetaData
{
    public class CTS220_DOSelectProcess_Meta
    {
        [NotNullOrEmpty(Controller = MessageUtil.MODULE_CONTRACT, Screen = "CTS220", Parameter = "lblOccurence", ControlName = "Occurence")]
        public string OCC { get; set; }
    }

    public class CTS220_DOSubContract_Meta
    {
        [NotNullOrEmpty(Controller = MessageUtil.MODULE_CONTRACT, Screen = "CTS220", Parameter = "lblSubContractorName", ControlName = "SubcontractCode")]
        public string SubContractCode { get; set; }
    }

    public class CTS220_ValidateRequireField_MetaData
    {
        [NotNullOrEmpty(Controller = MessageUtil.MODULE_CONTRACT,
            Screen = "CTS220",
            Parameter = "lblChangeOperationDate",
            ControlName = "ChangeImplementDate")]
        public DateTime? ChangeImplementDate { get; set; }

        [NotNullOrEmpty(Controller = MessageUtil.MODULE_CONTRACT,
            Screen = "CTS220",
            Parameter = "lblSalesMan1",
            ControlName = "SaleManEmpNo1")]
        public string SalesmanEmpNo1 { get; set; }
    }

    public class CTS220_ValidateChangeReasonType_MetaData
    {
        [NotNullOrEmpty(Controller = MessageUtil.MODULE_CONTRACT,
            Screen = "CTS220",
            Parameter = "lblReason",
            ControlName = "ChangeReasonType")]
        public string ChangeReasonType { get; set; }
    }
}

File: SECOM_AJIS/Scripts/Contract/CTS220.js

$(document).ready(function () {
});

function InitModeSection(steps) {
    $("#divSpecifyProcessTypeSection").SetEnableView(false);
//    $("#divSpecifyProcessTypeSection").attr('disabled', true);
//    $("#divSpecifyProcessTypeSection").attr('readonly', true);
//    $('#divSpecifyProcessTypeSection :input').attr('disabled', true);
//    $('#divSpecifyProcessTypeSection :input').attr('readonly', true);

    CallMultiLoadScreen("Contract", steps, MaintainScreenItem);
}

function MaintainScreenItem() {
    if (processType == "Delete")
        MaintainScreenItemDeleteMode();
    else
        MaintainScreenItemInsertMode();
}

function MaintainScreenItemInsertMode() {
//    $("#divSpecifyProcessTypeSection").attr('disabled', true);
//    $("#divSpecifyProcessTypeSection").attr('readonly', true);
//    $('#divSpecifyProcessTypeSection :input').attr('disabled', true);
//    $('#divSpecifyProcessTypeSection :input').attr('readonly', true);

//    $("#divContractBasicSection").show();
    $("#divProductSection").show();
//    $("#divChangeContractSection").show();
//    $("#divInsuranceSection").show();
//    $("#divContractDocumentSection").show();
//    $("#divFutureDateSection").show();
//    $("#divQuotationSection").show();
//    $("#divInstallationSection").show();

    $("#ChangeImplementDate").InitialDate();
    $("#ExpectedResumeDate").InitialDate();
    $("#ReturnToOriginalFeeDate").InitialDate();
    $("#InstallationCompleteDate").InitialDate();

    SetRegisterCommand(true, BtnRegisterClick);
    SetResetCommand(true, BtnResetClick);
}

function MaintainScreenItemDeleteMode() {
//    $("#divSpecifyProcessTypeSection").attr('disabled', true);
//    $("#divSpecifyProcessTypeSection").attr('readonly', true);
//    $('#divSpecifyProcessTypeSection :input').attr('disabled', true);
//    $('#divSpecifyProcessTypeSection :input').attr('readonly', true);

//    $("#divContractBasicSection").show();
//    $("#divContractBasicSection").attr('disabled', true);
//    $("#divContractBasicSection").attr('readonly', true);
//    $('#divContractBasicSection :input').attr('disabled', true);
//    $('#divContractBasicSection :input').attr('readonly', true);
    $('#divContractBasicSection').SetDisabled(true);

//    $("#divProductSection").show();
//    $("#divProductSection").attr('disabled', true);
//    $("#divProductSection").attr('readonly', true);
//    $('#divProductSection :input').attr('disabled', true);
//    $('#divProductSection :input').attr('readonly', true);
    $('#divProductSection').SetDisabled(true);

//    $("#divChangeContractSection").show();
//    $("#divChangeContractSection").attr('disabled', true);
//    $("#divChangeContractSection").attr('readonly', true);
//    $('#divChangeContractSection :input').attr('disabled', true);
//    $('#divChangeContractSection :input').attr('readonly', true);
    $('#divChangeContractSection').SetDisabled(true);

//    $("#divInsuranceSection").show();
//    $("#divInsuranceSection").attr('disabled', true);
//    $("#divInsuranceSection").attr('readonly', true);
//    $('#divInsuranceSection :input').attr('disabled', true);
//    $('#divInsuranceSection :input').attr('readonly', true);
    $('#divInsuranceSection').SetDisabled(true);

//    $("#divContractDocumentSection").show();
//    $("#divContractDocumentSection").attr('disabled', true);
//    $("#divContractDocumentSection").attr('readonly', true);
//    $('#divContractDocumentSection :input').attr('disabled', true);
//    $('#divContractDocumentSection :input').attr('readonly', true);
    $('#divContractDocumentSection').SetDisabled(true);

//    $("#divFutureDateSection").show();
//    $("#divFutureDateSection").attr('disabled', true);
//    $("#divFutureDateSection").attr('readonly', true);
//    $('#divFutureDateSection :input').attr('disabled', true);
//    $('#divFutureDateSection :input').attr('readonly', true);
    $('#divFutureDateSection').SetDisabled(true);

//   $("#divQuotationSection").show();
//    $("#divQuotationSection").attr('disabled', true);
//    $("#divQuotationSection").attr('readonly', true);
//    $('#divQuotationSection :input').attr('disabled', true);
//    $('#divQuotationSection :input').attr('readonly', true);
    $('#divQuotationSection').SetDisabled(true);

//    $("#divInstallationSection").show();
//    $("#divInstallationSection").attr('disabled', true);
//    $("#divInstallationSection").attr('readonly', true);
//    $('#divInstallationSection :input').attr('disabled', true);
//    $('#divInstallationSection :input').attr('readonly', true);
    $('#divInstallationSection').SetDisabled(true);

    //Disable mygridCTS220_09
    if (CheckFirstRowIsEmpty(mygridCTS220_09, false) == false) {
        BindInstallationGrid(false);
    }

    SetOKCommand(true, BtnOKClick);
    SetCancelCommand(true, BtnCancelClick);

    SetRegisterCommand(false, null);
    SetResetCommand(false, null);
}

function BtnRegisterClick() {
    command_control.CommandControlMode(false);

    obj = GetValidateObject();
    call_ajax_method_json('/Contract/RegisterClick_CTS220', obj,
        function (result, controls) {
            if (controls != undefined) {
                VaridateCtrl(["OrderContractFee",
                            "ChangeType",
                            "ChangeImplementDate",
                            "SaleManEmpNo1",
                            "SaleManEmpNo2",
                            "NegotiationStaffEmpNo1",
                            "NegotiationStaffEmpNo2",
                            "ChangeReasonType",
                            "ReturnToOriginalFeeDate",
                            "ExpectedResumeDate"], controls);
                return;
            }
            else if (result != undefined) {
                $("#divContractBasicSection").SetViewMode(true);

                //$("#divProductSection").SetViewMode(true);
                SetSectionModeCTS220_03(true);

                //$("#divChangeContractSection").SetViewMode(true);
                SetSectionModeCTS220_04(true);

                $("#divInsuranceSection").SetViewMode(true);
                $("#divContractDocumentSection").SetViewMode(true);
                $("#divFutureDateSection").SetViewMode(true);
                $("#divQuotationSection").SetViewMode(true);

                //$("#divInstallationSection").SetViewMode(true);
                SetSectionModeCTS220_09(true);

                SetConfirmCommand(true, BtnConfirmClick);
                SetBackCommand(true, BtnBackClick);
                SetRegisterCommand(false, null);
                SetResetCommand(false, null);
            }
        });

    //command_control.CommandControlMode(true);
}

function BtnResetClick() {
    DisableResetCommand(true);

    /* --- Get Message --- */
    var obj = {
        module: "Common",
        code: "MSG0038"
    };
    call_ajax_method_json("/Shared/GetMessage", obj, 
        function (result) {
            OpenOkCancelDialog(result.Code, result.Message,
            function () {
    //            InitialSelectProcessMode();
    //            $("#rdoCorrectSecurityData").attr("checked", true);
    //            $("#Occurence").val("");

    //            SetRegisterCommand(false, null);
    //            SetResetCommand(false, null);

                //SetResetState();
                var obj = { ContractCode: $("#ContractCode").val() };
                ajax_method.CallScreenControllerWithAuthority("/Contract/CTS220", obj, false, null);
            },
            null);
        });

    DisableResetCommand(false);
}

function SetResetState() {
    InitialSelectProcessMode();
    $("#rdoCorrectSecurityData").attr("checked", true);
    $("#Occurence").val("");

    SetRegisterCommand(false, null);
    SetResetCommand(false, null);
    SetConfirmCommand(false, null);
    SetBackCommand(false, null);
    SetOKCommand(false, null);
    SetCancelCommand(false, null);
}

function BtnConfirmClick() {
    command_control.CommandControlMode(false);

    obj = GetValidateObject();
    call_ajax_method_json('/Contract/ConfirmClick_CTS220', obj,
        function (result, controls) {
            //if (result.Code == "MSG0046") {
            if (result != undefined) {
                OpenInformationMessageDialog(result.Code, result.Message, function () {
                    //SetConfirmCommand(false, null);
                    //SetBackCommand(false, null);
                    //BtnResetClick();
                    SetResetState();
                }, null);
            }
        });

    //command_control.CommandControlMode(true);
}

function BtnBackClick() {
    SetConfirmCommand(false, null);
    SetBackCommand(false, null);

    SetRegisterCommand(true, BtnRegisterClick);
    SetResetCommand(true, BtnResetClick);

    $("#divContractBasicSection").SetViewMode(false);
    $("#ImportantFlag").attr("disabled", true)

    //$("#divProductSection").SetViewMode(false);
    SetSectionModeCTS220_03(false);

    //$("#divChangeContractSection").SetViewMode(false);
    SetSectionModeCTS220_04(false);

    $("#divInsuranceSection").SetViewMode(false);
    $("#divContractDocumentSection").SetViewMode(false);
    $("#divFutureDateSection").SetViewMode(false);
    $("#divQuotationSection").SetViewMode(false);

    //$("#divInstallationSection").SetViewMode(false);
    SetSectionModeCTS220_09(false);
}

function BtnOKClick() {
    obj = GetValidateObject();
    call_ajax_method('/Contract/OKClick_CTS220', obj,
    function (result, controls) {
        if (result == undefined) {

            var obj = {
                module: "Contract",
                code: "MSG3137"
            };

            call_ajax_method_json("/Shared/GetMessage", obj, 
                function (result) {
                    OpenYesNoWarningMessageDialog(result.Code, result.Message,
                        function () {
                            obj = GetValidateObject();
                            call_ajax_method('/Contract/ConfirmClick_CTS220', obj,
                            function (result, controls) {
//                                if (result == true) {
//                                    OpenErrorMessageDialog(result.Code, result.Message, function () { }, null);
//                                }

//                                if (result != true && result != false) {
                                if (result != undefined) {
                                    OpenInformationMessageDialog(result.Code, result.Message, function () {
                                        //SetOKCommand(false, null);
                                        //SetCancelCommand(false, null);
                                        //BtnResetClick();
                                        SetResetState();
                                    }, null);
                                }
                            });
                        });
                });
        }
    });
}

function BtnCancelClick() {
//    InitialSelectProcessMode();
//    $("#rdoCorrectSecurityData").attr("checked", true);
//    $("#Occurence").val("");

//    SetOKCommand(false, null);
//    SetCancelCommand(false, null);
    SetResetState();
}

function GetValidateObject() {
    var processType;
    if ($("#rdoCorrectSecurityData").prop("checked") == true)
        processType = "Correct";

    if ($("#rdoInsertSecurityOccurence").prop("checked") == true)
        processType = "Insert";

    if ($("#rdoDeleteOperatedOccurence").prop("checked") == true)
        processType = "Delete";

    var operationType = new Array();
    $("#OperationType").find("input:checkbox").each(function () {
        if ($(this).prop("checked") == true) {
            operationType.push($(this).val());
        }
    });

    var obj = {
        ContractCode: $("#ContractCode").val(),
        OCC: $("#Occurence").val(),
        ProcessType: processType,

        ContractFeeCurrencyType: $("#OrderContractFee").NumericCurrencyValue(),
        ContractFee: $("#OrderContractFee").NumericValue(),
        ContractFeeOnStopCurrencyType: $("#StopFee").NumericCurrencyValue(),
        ContractFeeOnStop: $("#StopFee").NumericValue(), //Add by Jutarat A. on 14082012

        ChangeImplementDate: $("#ChangeImplementDate").val(),
        ChangeType: $("#ChangeType").val(),
        ChangeReasonType: $("#ChangeReasonType").val(),
        DocumentCode: $("#DocumentName").val(),
        SecurityMemo: $("#SecurityMemo").val(),
        //OperationTypeCode: $("#OperationTypeCode").val(),
        InsuranceTypeCode: $("#InsuranceType").val(),

        InsuranceCoverageAmountCurrencyType: $("#InsuranceCoverageFee").NumericCurrencyValue(),
        InsuranceCoverageAmount: $("#InsuranceCoverageFee").NumericValue(),
        MonthlyInsuranceFeeCurrencyType: $("#MonthlyInsuranceFee").NumericCurrencyValue(),
        MonthlyInsuranceFee: $("#MonthlyInsuranceFee").NumericValue(),

        PlanCode: $("#PlanCode").val(),

        MaintenanceFee1CurrencyType: $("#MaintenanceFee1").NumericCurrencyValue(),
        MaintenanceFee1: $("#MaintenanceFee1").NumericValue(),
        AdditionalFee1CurrencyType: $("#AdditionalFee1").NumericCurrencyValue(),
        AdditionalFee1: $("#AdditionalFee1").NumericValue(),
        AdditionalFee2CurrencyType: $("#AdditionalFee2").NumericCurrencyValue(),
        AdditionalFee2: $("#AdditionalFee2").NumericValue(),
        AdditionalFee3CurrencyType: $("#AdditionalFee3").NumericCurrencyValue(),
        AdditionalFee3: $("#AdditionalFee3").NumericValue(),

        InstallationTypeCode: $("#InstallationTypeCode").val(),
        InstallationCompleteDate: $("#InstallationCompleteDate").val(),

        NormalInstallFeeCurrencyType: $("#NormalInstallFee").NumericCurrencyValue(),
        NormalInstallFee: $("#NormalInstallFee").NumericValue(),
        OrderInstallFeeCurrencyType: $("#OrderInstallFee").NumericCurrencyValue(),
        OrderInstallFee: $("#OrderInstallFee").NumericValue(),
        OrderInstallFee_ApproveContractCurrencyType: $("#OrderInstallFee_ApproveContract").NumericCurrencyValue(),
        OrderInstallFee_ApproveContract: $("#OrderInstallFee_ApproveContract").NumericValue(),
        OrderInstallFee_CompleteInstallCurrencyType: $("#OrderInstallFee_CompleteInstall").NumericCurrencyValue(),
        OrderInstallFee_CompleteInstall: $("#OrderInstallFee_CompleteInstall").NumericValue(),
        OrderInstallFee_StartServiceCurrencyType: $("#OrderInstallFee_StartService").NumericCurrencyValue(),
        OrderInstallFee_StartService: $("#OrderInstallFee_StartService").NumericValue(),

        ReturnToOriginalFeeDate: $("#ReturnToOriginalFeeDate").val(),
        ExpectedResumeDate: $("#ExpectedResumeDate").val(),
        SalesmanEmpNo1: $("#SaleManEmpNo1").val(),
        SalesmanEmpNo2: $("#SaleManEmpNo2").val(),
        NegotiationStaffEmpNo1: $("#NegotiationStaffEmpNo1").val(),
        NegotiationStaffEmpNo2: $("#NegotiationStaffEmpNo2").val(),
        ApproveNo1: $("#ApproveNo1").val(),
        ApproveNo2: $("#ApproveNo2").val(),
        IsEnableReasonType: ($("#ChangeReasonType").prop("disabled") == false),
        OperationType: operationType,

        NormalContractFeeCurrencyType: $("#NormalContractFee").NumericCurrencyValue(),
        NormalContractFee: $("#NormalContractFee").NumericValue(), //Add by Jutarat A. on 06022014

        InstallFeePaidBySECOMCurrencyType: $("#InstallFeePaidBySECOM").NumericCurrencyValue(),
        InstallFeePaidBySECOM: $("#InstallFeePaidBySECOM").NumericValue(),
        InstallFeeRevenueBySECOMCurrencyType: $("#InstallFeeRevenueBySECOM").NumericCurrencyValue(),
        InstallFeeRevenueBySECOM: $("#InstallFeeRevenueBySECOM").NumericValue()
    }

    return obj;
}
